pkg/reconciliation/construct_podtemplatespec.go

```diff
@@ -710,6 +710,7 @@ func buildContainers(dc *api.CassandraDatacenter, baseTemplate *corev1.PodTempla
 		{Name: "CLUSTER_NAME", Value: dc.Spec.ClusterName},
 		{Name: "DATACENTER_NAME", Value: dc.DatacenterName()},
 		{Name: "RACK_NAME", ValueFrom: selectorFromFieldPath("metadata.labels['cassandra.datastax.com/rack']")},
+		{Name: "NAMESPACE", Value: dc.Namespace},
```

Comment:
I don't think this is a correct approach in general sense or how other stuff is done. The pod itself knows the namespace already, it's in the file ``/var/run/secrets/kubernetes.io/serviceaccount/namespace`` so just catting that would be enough. This is also something that can be used in the upstream alone without needing a new version of cass-operator by simply modifying the vector configuration there.

The other "correct" method I think is to use selector, so:

```go
{Name: "NAMESPACE", ValueFrom: selectorFromFieldPath("metadata.namespace")},
```

Reply:
Vector VRLs language doesn't have a function to read data from a file. All we can do is read the hostname or read an env variable (see [here](https://vector.dev/docs/reference/vrl/functions/#system-functions)).
So I think we need the additional variable to avoid building a convoluted solution.
I'm ok with using the "metadata.namespace" instead of the dc namespace, I'll push a change for this.